lib/tasks/posts.rake

```diff
@@ -450,14 +452,17 @@ task "posts:reorder_posts", [:topic_id] => [:environment] do |_, args|
 
     builder = DB.build <<~SQL
       UPDATE
-        posts
+        posts AS p
       SET
         post_number = sort_order
+      FROM
+        topics t
       /*where*/
     SQL
 
-    builder.where("topic_id = ?", args[:topic_id]) if args[:topic_id]
-    builder.where("post_number < 0")
+    builder.where("t.id = p.topic_id")
```

Comment:
is there a chance that args.topic_id is blank/null? See clause below.

Reply:
Yes, the rake task can be executed with a specific topic or without, if executed without, it will run for all Posts